platform/darwin/src/MGLStyle.mm

```diff
@@ -205,12 +206,21 @@ - (void)insertLayer:(MGLStyleLayer *)layer belowLayer:(MGLStyleLayer *)otherLaye
 
 - (void)addSource:(MGLSource *)source
 {
-    self.mapView.mbglMap->addSource(source.mbglSource);
+    // After it is added, the pending source is moved to the mbgl map and is
+    // no longer "pending". The move operation sets it to NULL. However, the
+    // MGLSource's rawSource will still point to the mbgl source and clients
+    // can mutate it via that reference.
+    self.mapView.mbglMap->addSource(source.pendingSource);
```

Comment:
> The pending source pointer exists only to be transferred to the mbgl object.

To me, that suggests we may want to go with the approach I outlined above. Not only does it make it clear where ownership is transferred, but it also prevents misuse of the pendingSource pointer. Currently you can use that property however you like.